src/mon/OSDMonitor.cc

```diff
@@ -1087,6 +1087,11 @@ void OSDMonitor::encode_pending(MonitorDBStore::TransactionRef t)
 
     // clean inappropriate pg_upmap/pg_upmap_items (if any)
     osdmap.maybe_remove_pg_upmaps(cct, osdmap, tmp, &pending_inc);
+    // throw out any stale upmap items too
+    // note that the above check may have deleted some upmap items already
+    // it shouldn't matter here since clean_pg_upmaps is basically doing
+    // the same (e.g., re-deleting the same item is generally ok)
+    tmp.clean_pg_upmaps(cct, &pending_inc);
```

Comment:
should we collapse these two functions into one, since this is now the only real call site?

Reply:
@liewegas 
I repushed a quick fix for now (I have some other errors to investigate). Will probably do a more complete fix (e.g., merge both__maybe_remove_pg_upmaps__ and __clean_pg_upmaps__ into a single __clean_pg_upmaps__ function) in a follow-up pr.